scripts/go_install.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [ -z "${1}" ]; then
+  echo "must provide module as first parameter"
+  exit 1
+fi
+
+if [ -z "${2}" ]; then
+  echo "must provide binary name as second parameter"
+  exit 1
+fi
+
+if [ -z "${3}" ]; then
+  echo "must provide version as third parameter"
+  exit 1
+fi
+
+if [ -z "${GOBIN}" ]; then
+  echo "GOBIN is not set. Must set GOBIN to install the bin in a specified directory."
+  exit 1
+fi
+
+tmp_dir=$(mktemp -d -t goinstall_XXXXXXXXXX)
+function clean {
+  rm -rf "${tmp_dir}"
+}
+trap clean EXIT
+
+rm "${GOBIN}/${2}"* || true
+
+cd "${tmp_dir}"
+
+# create a new module in the tmp directory
+go mod init fake/mod
+
+# install the golang module specified as the first argument
+go get -tags tools "${1}@${3}"
```

Comment:
Rather than using `go get` in a fake module, did you consider using `GOBIN=somedir go install pkg@version`? Then you wouldn't need the temp directory, and wouldn't mess with the global GOBIN dir.

Reply:
+1, [since go 1.17 installing executables with go get is deprecated](https://go.dev/doc/go-get-install-deprecation#:~:text=go%20get%20%E2%80%99s%20ability%20to%20build%20and%20install,that%20can%E2%80%99t%20be%20built%20for%20the%20current%20platform.)

Reply:
opened an issue in CAPZ to track the same change https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2004

Reply:
Opened an issue to use go install rather than go get
https://github.com/kubernetes-sigs/cluster-api/issues/6008